app/javascript/components/visualization/ScatterPlot.jsx

```diff
@@ -941,3 +923,70 @@ function getDragMode(isCellSelecting) {
 }
 
 ScatterPlot.getPlotlyTraces = getPlotlyTraces
+
+/** Intersect filtered cells with dataArrays results */
+export function intersect(filteredCells, scatter) {
+  const intersectedData = {}
+  const dataArrays = scatter.data
+  const keys = Object.keys(dataArrays)
+  keys.forEach(key => intersectedData[key] = [])
+  window.dataArrays = dataArrays
+  // array of cell indexes that are being plotted
+  const plotted = []
+  for (let i = 0; i < filteredCells.length; i++) {
+    const filteredCell = filteredCells[i]
+    const allCellsIndex = filteredCell.allCellsIndex
+    plotted.push(allCellsIndex)
+    for (let j = 0; j < keys.length; j++) {
+      const key = keys[j]
+      const dataArray = dataArrays[key]
+      const filteredElement = dataArray[allCellsIndex]
+      intersectedData[keys[j]].push(filteredElement)
+    }
+  }
+  return [intersectedData, plotted]
+}
+
+ScatterPlot.intersect = intersect
+
+/**
+ * Reassign plot data to change cells filtered by cell faceting to --Filtered-- trace (light grey, transparent)
+ * this keeps the shape of the original plot but clearly shows what cells are excluded
+ *
+ * @param plotted {Array} Indices of points that are to be plotted
+ * @param originalData {Object} original scatter data object
+ * @param filteredData {Object} filtered scatter data object from cell faceting
+ * @returns {Object} reorganized scatter data object with new --Filtered-- trace
+ */
+export function reassignFilteredCells(plotted, originalData, filteredData) {
+  const allIndexes = [...Array(originalData['x'].length).keys()]
+  const originalSet = new Set(allIndexes)
+  const plottedSet = new Set(plotted)
+  const reassignedIndices = [...originalSet].filter(i => !plottedSet.has(i))
+  const newPlotData = {}
+  const keys = Object.keys(originalData)
+  keys.forEach(key =>  {
+    newPlotData[key] = []
+  })
+  for (let idx = 0; idx < reassignedIndices.length; idx++) {
+    for (let k = 0; k < keys.length; k++) {
+      const key = keys[k]
```

Comment:
Normally I'd agree, but in this particular case the outer loop is using `idx` as the counter variable, and I didn't want the inner one to use `i`.  I went for `k` since it's the iterator for the `keys`, whereas the outer loop is going through an array of indices.